### PR TITLE
load retrieval data

### PR DESCRIPTION
**Pull request by oahehc**
related: https://github.com/TaiBIF/camera-trap-vueapp/issues/80

image

image

image

### CODE REVIEW

File: src/pages/Project/ProjectInfo/charts/CameraBarChart.vue [hunk 1]
```diff
@@ -90,38 +101,41 @@ export default {
...
   },
-  methods: {
-    changeSelectedYear(year) {
-      if (year > new Date().getFullYear()) {
-        return;
-      }
-      this.selectedYear = year;
+  watch: {
+    retrievalLoadingStatus: function(state) {
+      if (state === 'loaded') this.loadCameraChart();
```
Comment (oahehc) [inline src/pages/Project/ProjectInfo/charts/CameraBarChart.vue]: API loaded 後再更新 highchart

File: src/store/modules/projects/projects.js [hunk 2]
```diff
@@ -65,6 +105,24 @@ const mutations = {
...
+
+        return {
+          ...res,
+          [id]: data,
+        };
+      },
+      state.retrievalData[year] || {},
+    );
+
+    Vue.set(state.retrievalData, year, selectedYearData);
```
Comment (oahehc) [inline src/store/modules/projects/projects.js]: retrievalData: {
	2018: {
      [cameraLocationId]: [...],
      [studyAreaId]: [...],
    },
	...
}

**Comment by cjk-halodule** [comment]
Functionality verified. Conflicts to be resolved @oahehc

**Comment by oahehc** [comment]
fixed conflict

**Pull request closed without merging**